test/unit/math/prim/fun/to_int_test.cpp

```diff
@@ -0,0 +1,41 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <cmath>
+#include <limits>
+
+TEST(MathFunctions, to_int) {
+  using stan::math::to_int;
+
+  EXPECT_EQ(2, to_int(2.0));
+  EXPECT_EQ(2, to_int(2.1));
+  EXPECT_EQ(2, to_int(2.9));
+  EXPECT_EQ(2, to_int(2.999999999));
+
+  EXPECT_EQ(-36574, to_int(-36574.0));
+  EXPECT_EQ(-36574, to_int(-36574.1));
+  EXPECT_EQ(-36574, to_int(-36574.9));
+  EXPECT_EQ(-36574, to_int(-36574.999999999));
+
+  EXPECT_THROW(to_int(std::numeric_limits<int>::max() + 1.0),
+               std::invalid_argument);
+  EXPECT_THROW(to_int(std::numeric_limits<int>::min() - 1.0),
+               std::invalid_argument);
+}
+
+TEST(MathFunctions, to_int_vec) {
+  using stan::math::to_int;
+
```

Comment:
test empty vectors and size 1 vectors (this should all work given the way it's defined, but the boundary tests should still be in place).

I'm not sure we really need to test bigger containers or other types of containers like `vector<MatrixXd>`, etc.